tests/performance-tests/src/utils/publicApiService.ts

```diff
@@ -0,0 +1,290 @@
+  eq?: T;
+  notEq?: T;
+  gte?: T;
+  gt?: T;
+  lte?: T;
+  lt?: T;
+  in?: T[];
+  notIn?: T[];
+}
+
+export interface DataSetQueryCriteria {
+  filters?: DataSetQueryIdPredicate;
+  locations?: DataSetQueryIdPredicate;
+  parentLocations?: DataSetQueryIdPredicate;
+  geographicLevels?: DataSetQueryIdPredicate;
+  timePeriods?: DataSetQueryComparablePredicate<TimePeriod>;
+}
+
+export interface DataSetQueryConditionAnd {
+  and: DataSetQueryNode[];
+}
+
+export interface DataSetQueryConditionOr {
+  or: DataSetQueryNode[];
+}
+
+export interface DataSetQueryConditionNot {
+  not: DataSetQueryNode[];
+}
+
+export type DataSetQueryNode =
+  | DataSetQueryConditionAnd
+  | DataSetQueryConditionOr
+  | DataSetQueryConditionNot
+  | DataSetQueryCriteria;
+
+interface TimePeriod {
+  code: string;
+  year: number;
+}
+
+export interface DataSetQueryRequest {
+  facets: DataSetQueryNode;
+  indicators?: string[];
+  sort?: [
+    {
+      name: string;
+      order: string;
+    },
+  ];
+}
+
+export interface DataSetQueryResponse {
+  footnotes: [
+    {
+      id: string;
+      content: string;
+    },
+  ];
+  paging: {
+    page: 0;
+    pageSize: 0;
+    totalResults: 0;
+    totalPages: 0;
+  };
+  results: {
+    id: string;
+  }[];
+}
+
+// Some Publications and their data sets are currently not discoverable via
+// the API, so we can include them manually here.
+const hiddenPublications: Publication[] = [
+  {
+    id: '1681557f-510f-446e-bc9a-f2c7a59d1cfa',
+    slug: 'benchmark-publication',
+    title: 'Benchmark Publication',
+  },
+];
+
+const hiddenDataSets: { [publicationId: string]: { dataSets: DataSet[] } } = {
+  '1681557f-510f-446e-bc9a-f2c7a59d1cfa': {
+    dataSets: [
+      {
+        id: 'a96044e5-2310-4890-a601-8ca0b67d2964',
+        name: 'QUA01',
+        content: '',
+        geographicLevels: ['National'],
+        timePeriods: {
+          start: '2013/14',
+          end: '2018/19',
+        },
+      },
+    ],
+  },
+};
+
+class PublicApiService {
+  private readonly client: HttpClient;
+
+  constructor(baseUrl: string, checkResponseStatus = true) {
+    this.client = new HttpClient({
+      baseUrl,
+      checkResponseStatus,
+    });
+  }
+
+  listPublications() {
+    const { response, json } = this.client.get<Paged<Publication>>(
+      '/v1/publications?pageSize=40',
+    );
+    const publications: Publication[] = [
+      ...json.results,
+      ...hiddenPublications,
+    ];
+    return {
+      publications,
+      response,
+    };
+  }
+
+  listDataSets(publicationId: string) {
+    let response: RefinedResponse<'text'> | undefined;
+    let json: DataSet[];
+
+    if (Object.keys(hiddenDataSets).includes(publicationId)) {
+      json = hiddenDataSets[publicationId].dataSets;
+    } else {
+      const result = this.client.get<DataSet[]>(
+        `/v1/publications/${publicationId}/data-sets`,
+      );
+      response = result.response;
+      json = result.json;
```

Comment:
Could use destructuring:

```ts
const { response, json } = this.client.get(...);
```

Reply:
Cool will do.  Almost surprised the linter didn't pull this up!

Reply:
Yeah I figured there was a reason this didn't get pulled up by the linter.  It's cos these are declared with `let` above.